solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -438,81 +436,78 @@ public void send(OutStream outStream, SolrRequest<?> req, String collection) thr
   private static final Cancellable FAILED_MAKING_REQUEST_CANCELLABLE = () -> {};
 
   public Cancellable asyncRequest(
-      SolrRequest<?> solrRequest,
-      String collection,
-      AsyncListener<NamedList<Object>> asyncListener) {
+      SolrRequest<?> solrReq, String collection, AsyncListener<NamedList<Object>> asyncListener) {
+    MDCCopyHelper mdcCopyHelper = new MDCCopyHelper();
+    SolrRequest<?> solrRequest = unwrapV2Request(solrReq);
+
     Request req;
     try {
-      req = makeRequest(solrRequest, collection);
-    } catch (SolrServerException | IOException e) {
-      asyncListener.onFailure(e);
-      return FAILED_MAKING_REQUEST_CANCELLABLE;
-    }
-    final ResponseParser parser =
-        solrRequest.getResponseParser() == null ? this.parser : solrRequest.getResponseParser();
-    MDCCopyHelper mdcCopyHelper = new MDCCopyHelper();
-    req.onRequestQueued(asyncTracker.queuedListener)
-        .onComplete(asyncTracker.completeListener)
-        .send(
-            new InputStreamResponseListener() {
-              @Override
-              public void onHeaders(Response response) {
-                super.onHeaders(response);
-                InputStreamResponseListener listener = this;
-                executor.execute(
-                    () -> {
-                      InputStream is = listener.getInputStream();
-                      assert ObjectReleaseTracker.track(is);
-                      try {
-                        NamedList<Object> body =
-                            processErrorsAndResponse(
-                                solrRequest, parser, response, is, req.getURI().toString());
+      String url = getRequestPath(solrRequest, collection);
+      InputStreamResponseListener listener =
+          new InputStreamResponseListener() {
+            @Override
+            public void onHeaders(Response response) {
+              super.onHeaders(response);
+              InputStreamResponseListener listener = this;
```

Comment:
must this be defined a second time?

Reply:
does not seem to be needed. weird, removed it now